node/pkg/vaa/structs_test.go

```diff
@@ -0,0 +1,117 @@
+package vaa
+
+import (
+	"encoding/hex"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestChainIDFromString(t *testing.T) {
+	type test struct {
+		i string
+		o ChainID
+	}
+
+	// Positive Test Cases
+	p_tests := []test{
+		{i: "solana", o: ChainIDSolana},
+		{i: "ethereum", o: ChainIDEthereum},
+		{i: "terra", o: ChainIDTerra},
+		{i: "bsc", o: ChainIDBSC},
+		{i: "polygon", o: ChainIDPolygon},
+		{i: "avalanche", o: ChainIDAvalanche},
+		{i: "oasis", o: ChainIDOasis},
+		{i: "fantom", o: ChainIDFantom},
+		{i: "algorand", o: ChainIDAlgorand},
+		{i: "ethereum-ropsten", o: ChainIDEthereumRopsten},
+		{i: "Solana", o: ChainIDSolana},
+		{i: "Ethereum", o: ChainIDEthereum},
+		{i: "Terra", o: ChainIDTerra},
+		{i: "Bsc", o: ChainIDBSC},
+		{i: "Polygon", o: ChainIDPolygon},
+		{i: "Avalanche", o: ChainIDAvalanche},
+		{i: "Oasis", o: ChainIDOasis},
+		{i: "Fantom", o: ChainIDFantom},
+		{i: "Algorand", o: ChainIDAlgorand},
+		{i: "Karura", o: ChainIDKarura},
+		{i: "Acala", o: ChainIDAcala},
```

Comment:
Uber minor nit here, but `i` and `o` are pretty bad names. The variable `i` is generally used to denote a loop counter, and these are strings. In the future, can you use slightly more descriptive names?